### PR TITLE
[stable/drone] add gitea oauth2 support

### PR DESCRIPTION
**Pull request by cgroschupp**
#### What this PR does / why we need it:
Drone no longer supports basic auth since 1.4.0, see at https://github.com/drone/drone/pull/2721

https://docs.drone.io/installation/providers/gitea/#step-3-configuration

#### Checklist
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

File: stable/drone/Chart.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ apiVersion: v1
 name: drone
 home: https://drone.io/
 icon: https://drone.io/apple-touch-icon.png
-version: 2.3.1
```
Comment (zakkg3) [inline stable/drone/Chart.yaml]: version now is version: 2.3.2

**Comment by cgroschupp** [comment]
@zakkg3 Do you know why this pr is not merged?

**Pull request merged**